### PR TITLE
Team Members

### PR DESCRIPTION
**Pull request by GrahamCampbell**
I've created this pull request so all of us can easily track how this is going.

### CODE REVIEW

**Comment by jbrooksuk** [comment]
Thanks!

**Comment by kfrankperson** [comment]
:+1: Love this idea

**Comment by GrahamCampbell** [comment]
This will need rebasing at some point too.

**Comment by jbrooksuk** [comment]
Yup, will be doing this shortly.

File: public/build/css/all.css [hunk 1]
```diff
@@ -1,3 +1,7 @@
+/*!
```
Comment (GrahamCampbell) [inline public/build/css/all.css]: this keeps coming back?

Comment (jbrooksuk) [inline public/build/css/all.css]: ?

**Comment by GrahamCampbell** [comment]
`Rebased onto master` You need to `git pull --rebase origin master` then `git push --force`. Merge commits are bad. :(

File: public/build/css/all.css (hunk 1 above)
Comment (GrahamCampbell) [inline public/build/css/all.css]: When @joecohens compiles this file, you always see this header disappear, but when you compile it, it comes back? We're not using font awesome. Why is it still here?

Comment (jbrooksuk) [inline public/build/css/all.css]: Perhaps @joecohens is compiling with `gulp --production`? I should be doing that, in fact, it should be default.

**Comment by GrahamCampbell** [comment]
Rebasing this one now...

**Comment by jbrooksuk** [comment]
@GrahamCampbell more rebase?

**Comment by GrahamCampbell** [comment]
WIll do soon.

File: public/build/css/all.css (hunk 1 above)
Comment (joecohens) [inline public/build/css/all.css]: Yes I was compiling with gulp --production. I think we should minify before commiting.

**Comment by GrahamCampbell** [comment]
Rebased.

**Comment by GrahamCampbell** [comment]
Someone has pushed hetre and deleded all my changes?

**Comment by joecohens** [comment]
?

**Comment by GrahamCampbell** [comment]
I had force pushed here after rebasing this. There should only be 1 commit here, but someone has deleted it and pushed the old stuff.

**Comment by jbrooksuk** [comment]
Must've been me, but I don't remember doing it.

**Comment by GrahamCampbell** [comment]
Sure. It was only about 5 mins work. I'll do it again at some point, but
please don't delete it again, lol.

**Comment by jbrooksuk** [comment]
Sorry about that :( I don't know how I'd have managed it.

**Comment by jbrooksuk** [comment]
Just rebased everything, but now I'll squash and continue work.

**Comment by jbrooksuk** [comment]
There we go :smile: 

I'm hoping to get this done today - but I've got to go out shortly, and I'm not sure for how long.

File: src/Http/Controllers/DashboardController.php [hunk 2]
```diff
@@ -24,6 +24,21 @@ public function showDashboard()
     }
 
     /**
+     * Shows the team members view.
+     *
+     * @return \Illuminate\View\View
+     */
+    public function showTeamView()
+    {
+        $team = User::all();
```
Comment (GrahamCampbell) [inline src/Http/Controllers/DashboardController.php]: you need to import this

File: src/Models/User.php [hunk 3]
```diff
@@ -72,4 +72,14 @@ public function getGravatarAttribute($size = 200)
...
     }
+
+    /**
+     * Returns whether a user is at admin level.
+     *
+     * @return bool
+     */
+    public function getIsAdminAttribute()
+    {
+        return (int) $this->level === 1;
```
Comment (GrahamCampbell) [inline src/Models/User.php]: this is not correct

Comment (GrahamCampbell) [inline src/Models/User.php]: php is doing `$this->level === 1`, then casting it to an int

Comment (jbrooksuk) [inline src/Models/User.php]: ?

File: src/Http/Controllers/DashboardController.php (hunk 2 above)
Comment (jbrooksuk) [inline src/Http/Controllers/DashboardController.php]: Yeah, I'll add that now.

File: src/Models/User.php (hunk 3 above)
Comment (jbrooksuk) [inline src/Models/User.php]: So I don't need the `(int)` because it will be anyway.

File: src/Models/User.php [hunk 4]
```diff
@@ -72,4 +72,14 @@ public function getGravatarAttribute($size = 200)
...
     }
+
+    /**
+     * Returns whether a user is at admin level.
+     *
+     * @return bool
+     */
+    public function getIsAdminAttribute()
+    {
+        return $this->level === 1;
```
Comment (GrahamCampbell) [inline src/Models/User.php]: is $this->level not actually a string because eloquent will cast it?

Comment (jbrooksuk) [inline src/Models/User.php]: It's an int in the db.

Comment (GrahamCampbell) [inline src/Models/User.php]: Yeh, but I've had issues with things like this before where I thought they were ints, but they were actually strings. Maybe try var_dumping it to see what it actually is. :)

Comment (jbrooksuk) [inline src/Models/User.php]: Ok, will do :)

**Comment by jbrooksuk** [comment]
Just editing left to do.

File: src/Http/Controllers/DashboardController.php [hunk 5]
```diff
@@ -24,6 +27,64 @@ public function showDashboard()
...
+     */
+    public function postAddUser()
+    {
+        $user = User::create(Binput::all());
+
+        if ($user->isValid()) {
+            return Redirect::back()->with('created', $user->isValid());
+        } else {
+            return Redirect::back()
+                           ->withInput(Binput::except('password'))
```
Comment (GrahamCampbell) [inline src/Http/Controllers/DashboardController.php]: incorrect indentation here

Comment (GrahamCampbell) [inline src/Http/Controllers/DashboardController.php]: You should not align them just indent with an extra 4 spaces.

File: src/Models/User.php (hunk 4 above)
Comment (GrahamCampbell) [inline src/Models/User.php]: Actually, better way: http://3v4l.org/r8vo1. Just cast it to an int. Done.

Comment (GrahamCampbell) [inline src/Models/User.php]: *Just cast it to a bool. Done.

**Comment by GrahamCampbell** [comment]
Just force pushed here. Please don't delete the changes.

**Comment by jbrooksuk** [comment]
@GrahamCampbell if I rebase this, I do need to force push?

**Comment by jbrooksuk** [comment]
Also @GrahamCampbell I think I know why I kept overwriting branches. I didn't have `push.default` set...

**Comment by jbrooksuk** [comment]
I need to rebase this, but I'm confused because I did a merge rather than a rebase originally.

**Comment by GrahamCampbell** [comment]
You should do `git pull --rebase origin master` than `git push --force`.

**Comment by GrahamCampbell** [comment]
You seem to have pushed your old commits back up again. You need to hard reset to my changes first.

**Comment by jbrooksuk** [comment]
Huh? When did I manage that?

**Comment by jbrooksuk** [comment]
If I follow your instructions now, I should be ok?

**Comment by GrahamCampbell** [comment]
Just don't try to start working while your local branch doesn't actually match the remote changes. That's what causing the issues.

**Comment by jbrooksuk** [comment]
Ah I see. I'm pretty sure I always do a `git pull`?

**Comment by GrahamCampbell** [comment]
Git pull doesn't actually cut it here because the commits have changed. You need to do this:
```
git fetch --all
git reset --hard origin/master
```

**Comment by GrahamCampbell** [comment]
Or, rather, use the correct branch.

**Comment by GrahamCampbell** [comment]
`origin/team-members`

**Comment by GrahamCampbell** [comment]
If it's a pain, just delete the repo, and clone it again...

**Comment by jbrooksuk** [comment]
Ok, that makes sense. So in the event of you (or someone) pushing to a branch I'm working on, I need to fetch and reset.

I know `git pull` does a fetch and merge right? So we're just saying "no, use the latest checkout".

**Comment by GrahamCampbell** [comment]
Git pull doesn't cut it because it will leave your old commits there, even when they don't exist anymore on the remote.

**Comment by jbrooksuk** [comment]
Ok, so what do I need to do here to rebase?

**Comment by GrahamCampbell** [comment]
No, you've already overwritten this branch again... too late

**Comment by jbrooksuk** [comment]
Balls.

**Comment by jbrooksuk** [comment]
Do I need to redo this whole branch?

**Comment by GrahamCampbell** [comment]
Nah, I'll sort it...

**Comment by GrahamCampbell** [comment]
Ok, sorted this.

**Comment by jbrooksuk** [comment]
Phew! Thanks!

Happy to merge?

**Comment by GrahamCampbell** [comment]
Wait for scrutinizer to complete...

**Comment by GrahamCampbell** [comment]
Merged.

**Pull request merged**